integration-tests/tests/api/target/usage.spec.ts

```diff
@@ -2509,6 +2509,135 @@ test.concurrent(
+  async ({ expect }) => {
+    const { createOrg } = await initSeed().createOwner();
+    const { createProject } = await createOrg();
+    const { createTargetAccessToken, toggleTargetValidation, updateTargetValidationSettings } =
+      await createProject(ProjectType.Single);
+    const token = await createTargetAccessToken({});
+    await toggleTargetValidation(true);
+    await updateTargetValidationSettings({
+      excludedClients: [],
+      requestCount: 2,
+      percentage: 0,
+      breakingChangeFormula: BreakingChangeFormula.RequestCount,
+    });
+
+    const sdl = /* GraphQL */ `
+      type Query {
+        a: String
+        b: String
+        c: String
+      }
+    `;
+
+    const queryA = parse(/* GraphQL */ `
+      query {
+        a
+      }
+    `);
+    const queryB = parse(/* GraphQL */ `
+      query {
+        a
+        b
+      }
+    `);
+
+    function collectA() {
+      client.collectUsage()(
+        {
+          document: queryA,
+          schema,
+          contextValue: {
+            request,
+          },
+        },
+        {},
+      );
+    }
+    function collectB() {
+      client.collectUsage()(
+        {
+          document: queryB,
+          schema,
+          contextValue: {
+            request,
+          },
+        },
+        {},
+      );
+    }
+
+    const schema = buildASTSchema(parse(sdl));
+
+    const schemaPublishResult = await token
+      .publishSchema({
+        sdl,
+        author: 'Kamil',
+        commit: 'initial',
+      })
+      .then(res => res.expectNoGraphQLErrors());
+
+    expect(schemaPublishResult.schemaPublish.__typename).toEqual('SchemaPublishSuccess');
+
+    const usageAddress = await getServiceHost('usage', 8081);
+
+    const client = createHive({
+      enabled: true,
+      token: token.secret,
+      usage: true,
+      debug: false,
+      agent: {
+        logger: createLogger('debug'),
+        maxSize: 1,
+      },
+      selfHosting: {
+        usageEndpoint: 'http://' + usageAddress,
+        graphqlEndpoint: 'http://noop/',
+        applicationUrl: 'http://noop/',
+      },
+    });
+
+    const request = new Request('http://localhost:4000/graphql', {
+      method: 'POST',
+      headers: {
+        'x-graphql-client-name': 'integration-tests',
+        'x-graphql-client-version': '6.6.6',
+      },
+    });
+
+    collectA();
+    collectB();
+
+    await waitFor(8000);
+
+    // try to remove `Query.a`
+    const above = await token
+      .checkSchema(/* GraphQL */ `
+        type Query {
+          b: String
+          c: String
+        }
+      `)
+      .then(r => r.expectNoGraphQLErrors());
+
+    if (above.schemaCheck.__typename !== 'SchemaCheckError') {
+      throw new Error(`Expected SchemaCheckError, got ${above.schemaCheck.__typename}`);
+    }
+
+    expect(above.schemaCheck.errors).toEqual({
+      nodes: [
+        {
+          message: "Field 'a' was removed from object type 'Query'",
```

Comment:
Without your change, is it the case that 'a' would be allowed to be removed when it should not be?

Reply:
yeah